resteasy-client/src/main/java/org/jboss/resteasy/client/jaxrs/internal/ClientResponse.java

```diff
@@ -70,12 +71,38 @@ public void setClientConfiguration(ClientConfiguration configuration)
       this.processor = configuration;
    }
 
-   @Override
-   public Object getEntity()
-   {
-      abortIfClosed();
-      return super.getEntity();
-   }
+	@Override
+	public Object getEntity() {
+		abortIfClosed();
+		Object entity = super.getEntity();
+		if (entity != null) {
+			return entity;
+		}
+		//Check if the entity was previously fully consumed
+		try {
+			InputStream inputStream = getEntityStream();

```

Comment:
Unfortunately, the implementation of the finalizer mechanism is somehow bugged. I should go and verify this specific scenario and to tell the exact condition that would cause an error (that's why I've said "possibly a problem" ;-) ), but we've been suffering from similar issues, some of which led to making those other methods synchronized there. If you want to look at an interesting video on this topic, check https://www.youtube.com/watch?v=UrGP6pfb0H8